fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1145,15 +1139,14 @@ def assert_state(self, state: TrainingState) -> None:
 def cast_inputs_to_fp16(*args: Any, **kwargs: Any) -> Tuple[Any, Any]:
     """
     Cast any Tensors in *args or **kwargs to FP16.
-
-    Doesn't currently support Tensors nested inside containers (e.g., dict).
     """
-    kwarg_keys, flat_args = pack_kwargs(*args, **kwargs)
-    tensor_inputs, packed_non_tensor_inputs = split_non_tensors(flat_args)
-    tensor_inputs = tuple(t.half() if torch.is_floating_point(t) else t for t in tensor_inputs)
-    flat_args = unpack_non_tensors(tensor_inputs, packed_non_tensor_inputs)
-    args, kwargs = unpack_kwargs(kwarg_keys, flat_args)
-    return args, kwargs
+
+    def fn(x: torch.Tensor) -> torch.Tensor:
+        if x.dtype is torch.float32:
```

Comment:
how about `torch.is_floating_point(x)`?

Reply:
I initially did that but then I think there is a slight help not calling half() if it is already half()? Am I missing it?